tools/testinfra/runner/src/main.rs

```diff
@@ -73,45 +88,79 @@ fn main() -> Result<()> {
     // don't run anything when just listing
     if options.list {
         for test in tests {
-            println!("{}", test.name);
+            println!("{}", test_name(&test.target, &test.unit));
         }
         exit(0);
     }
 
+    // connect to DB when it is provided
+    let mut db = match options.conn {
+        None => None,
+        Some(ref uri) => Some(
+            Client::connect(&uri, NoTls)
+                .with_context(|| format!("Couldn't connect to specified test DB at '{}'", uri))?,
+        ),
+    };
+    let disabled = query_disabled(&mut db);
+
     // run tests in parallel (retries share the same thread)
     let total = tests.len();
-    eprintln!("Running {} tests...", total);
+    eprintln!("Found {} tests...", total);
     let _ = rayon::ThreadPoolBuilder::new()
         .num_threads(options.threads)
         .build_global();
     let mut tests: Vec<TestResult> = tests
         .into_par_iter()
         .map(|test| {
-            let test = test.run(options.retries);
+            let run = !disabled.contains(&(test.target.clone(), test.unit.clone()))
+                || options.run_disabled;
+            let test = if run {
+                test.run(options.retries)
+            } else {
+                TestResult {
```

Comment:
This is a little confusing - I'd prefer some kind of explicit "disabled" state instead of setting `attempts: 0` and logging as SKIP.
There are plenty of cases where we legitimately skip tests (for example, oss does not support `yum`, so skips all yum-specific tests), and it would be nice to explicitly differentiate between disabled and skipped

Reply:
Added a test status field to the test results, with an enum to make these cases more explicit.